src/plugins/pbjx/components/history/delegate.js

```diff
@@ -1,37 +1,138 @@
+    this.isDbValueSameAsNodeValue = this.isDbValueSameAsNodeValue.bind(this);
+  }
+
+  /**
+   * @param {Schema} schema - An instance of Schema to use to create requests.
+   */
+  componentWillUnmount(schema) {
+    this.dispatch(clearResponse(schema.getCurie()));
+  }
 
-export default (dispatch) => ({
   /**
    * Initializes the container.
    *
    * This is needed because state is not wiped out
    * when components/container are unmounted.
-   *
-   * @param {StreamId} streamId - An instance of StreamId
-   * @param {Schema} schema - An instance of Schema to use to create requests.
    */
-  handleInitialize: (streamId, schema) => {
-    dispatch(schema.createMessage().set('stream_id', streamId));
-  },
+  handleInitialize() {
+    const { node, schema } = this.ownProps;
+    const streamId = StreamId.fromString(`${node.schema().getCurie().getMessage()}.history:${node.get('_id')}`);
+    this.dispatch(schema.createMessage().set('stream_id', streamId).set('count', 10));
+  }
 
   /**
    * Handles load more events.
    *
-   * @param {StreamId} streamId - An instance of StreamId
-   * @param {Schema} schema - An instance of Schema to use to create requests.
    * @param {string} since - Return events since this time.
    */
-  handleLoadMore: (streamId, schema, since) => {
-    dispatch(schema.createMessage({
+  handleLoadMore(since) {
+    const { node, schema } = this.ownProps;
+    const streamId = StreamId.fromString(`${node.schema().getCurie().getMessage()}.history:${node.get('_id')}`);
+    this.dispatch(schema.createMessage({
+      count: 10,
       stream_id: streamId,
       since,
     }));
-  },
+  }
 
   /**
-   * @param {Schema} schema - An instance of Schema to use to create requests.
+   * Handles Revert.
+   *
+   * @param {array} selected
    */
-  componentWillUnmount: (schema) => {
-    dispatch(clearResponse(schema.getCurie()));
-  },
-});
+  handleRevert(selected) {
+    const { node, nodeRequest, formName, setBlocks } = this.ownProps;
+    const updatedNode = node.clone();
+    selected.forEach((item) => {
+      const { id, value } = item;
+      const field = node.schema().getField(id);
+      if (id === 'blocks') {
+        setBlocks(this.dispatch, formName, value);
+        return;
+      }
+      updatedNode.clear(id);
+      if (field.isASingleValue()) {
+        updatedNode.set(id, value);
+      }
+      if (field.isASet()) {
+        updatedNode.addToSet(id, value);
+      }
+      if (field.isAList()) {
+        updatedNode.addToList(id, value ? Array.from(value) : []);
+      }
+      if (field.isAMap()) {
+        /* eslint-disable no-restricted-syntax */
+        for (const [k, v] of Object.entries(value)) {
+          updatedNode.addToMap(id, k, v);
+        }
+      }
+    });
+
+    this.dispatch(updateReponseNode(updatedNode, nodeRequest.schema().getCurie()));
+
+    setTimeout(() => {
+      this.dispatch(change(formName, '_forceDirty', true));
+    }, 1000);
```

Comment:
@rsumilang dont think a whole second is necessary here, probably fine to just drop it at the bottom of the stack

Reply:
@efjacobson I can maybe change it to half a second but it needed something. Firing it immediately after did not alter the form.

Reply:
@rsumilang yeah redux form is like that, it needs to process the changes from changing the request node first. i expect 

```
setTimeout(() => this.dispatch(change(formName, '_forceDirty', true)));
```

will be fine